### PR TITLE
Fix `plot_intermediate_values` example.

### LINKED ISSUE #1033: `plot_intermediate_values` example code does not contain intermediate values.
The example code snippet embedded in the documentation for [`plot_intermediate_values`](https://github.com/optuna/optuna/blob/master/optuna/visualization/intermediate_values.py) doesn't call `Trial.report` and thus does not contain intermediate values. The example should be updated. See also https://github.com/optuna/optuna/pull/1003#issuecomment-599359742. 

Current documentation with empty plot: https://optuna.readthedocs.io/en/latest/reference/visualization.html#optuna.visualization.plot_intermediate_values

### PR DESCRIPTION
**Pull request by hvy**
Fixes #1033.

This PR takes over https://github.com/optuna/optuna/pull/1050.

### CODE REVIEW

**Comment by hvy** [comment]
Rendered result https://35138-122299416-gh.circle-artifacts.com/0/docs/build/html/reference/visualization.html#optuna.visualization.plot_intermediate_values.

File: docs/source/scripts/plot_intermediate_values.py [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+    return 2 * x - 4
+
 
 def objective(trial):
-    x = trial.suggest_uniform("x", -100, 100)
-    y = trial.suggest_categorical("y", [-1, 0, 1])
-    return x ** 2 + y
+    lr = trial.suggest_loguniform("lr", 1e-5, 1e-1)
+
+    x = _rng.uniform(0, 3)
```
[the remark this answers is not in the document]
Comment (hvy) [reply docs/source/scripts/plot_intermediate_values.py]: Thanks, changed as suggested. I intentionally started from different points first (e.g. weights in a neural net) but you have a point from a HPO point of view.

File: optuna/visualization/intermediate_values.py [hunk 2]
```diff
@@ -23,12 +23,31 @@ def plot_intermediate_values(study):
...
+            def df(x):
+                return 2 * x - 4
+
             def objective(trial):
-                x = trial.suggest_uniform('x', -100, 100)
-                y = trial.suggest_categorical('y', [-1, 0, 1])
-                return x ** 2 + y
+                lr = trial.suggest_loguniform("lr", 1e-5, 1e-1)
+
+                x = random.uniform(0, 3)
```
[the remark this answers is not in the document]
Comment (hvy) [reply optuna/visualization/intermediate_values.py]: Fixed.

**Comment by hvy** [comment]
PTAL. I also increased the number of trials so that the `TPESampler` doesn't just sample random points.

File: docs/source/scripts/plot_intermediate_values.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hvy) [reply docs/source/scripts/plot_intermediate_values.py]: New plot https://36032-122299416-gh.circle-artifacts.com/0/docs/build/html/reference/visualization.html#optuna.visualization.plot_intermediate_values.

[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply docs/source/scripts/plot_intermediate_values.py]: >  I intentionally started from different points first (e.g. weights in a neural net)

I see. Thank you for your explanation.

**Pull request merged**